library/ssl_msg.c

```diff
@@ -5431,8 +5431,11 @@ int mbedtls_ssl_read( mbedtls_ssl_context *ssl, unsigned char *buf, size_t len )
     n = ( len < ssl->in_msglen )
         ? len : ssl->in_msglen;
 
-    memcpy( buf, ssl->in_offt, n );
-    ssl->in_msglen -= n;
+    if ( buf )
```

Comment:
If `buf == NULL` but `len != 0`, that's a user error.
```suggestion
    if ( len != 0 )
```